R/get_oaproc.R

```diff
@@ -0,0 +1,289 @@
+#'
+#' }
+#'
+get_xs_points <- function(point1, point2, num_pts) {
+
+  point1 <- check_point(point1)[[1]]
+  point2 <- check_point(point2)[[1]]
+
+  url_base <- paste0(get_nldi_url(tier = "prod"), "/pygeoapi/processes/")
+
+  url <- paste0(url_base, "nldi-xsatendpts/jobs?response=document")
+
+  return(sf_post(url, make_json_input_xspts(point1, point2, num_pts)))
+
+}
+
+sf_post <- function(url, json) {
+  tryCatch({
+
+    out <- httr::RETRY("POST", url, httr::accept_json(),
+                       httr::content_type_json(),
+                       body = json)
+
+    if(out$status_code == 200) {
+      sf::read_sf(rawToChar(out$content))
+    } else {
+      stop(rawToChar(out$content))
+    }
+
+  }, error = function(e) {
+    message("Error calling processing service. \n Original error: \n", e)
+    NULL
+  })
+}
+
+
+check_point <- function(p) {
+  mess <- "Point must be of type sfc and have a CRS declared."
+
+  if(inherits(p, "sf")) p <- sf::st_geometry(p)
+
+  if(!inherits(p, "sfc")) stop(mess)
+
+  tryCatch({
+    sf::st_transform(p, 4326)
+  }, error = function(e) {
+    stop(paste(mess, "Original error was: \n", e))
+  })
+}
+
+make_json_input_trace <- function(p, raindrop = TRUE, direction = "down") {
+
+  jsonlite::toJSON(list(inputs = list(list(id = "lat",
+                                           type = "text/plain",
+                                           value = p[2]),
+                                      list(id = "lon",
+                                           type = "text/plain",
+                                           value = p[1]),
+                                      list(id = "raindroptrace",
+                                           type = "text/plain",
+                                           value = ifelse(raindrop,
+                                                          "true", "false")),
+                                      list(id = "direction",
+                                           type = "text/plain",
+                                           value = direction))),
+                   pretty = TRUE, auto_unbox = TRUE)
+}
+
+make_json_input_split <- function(p, upstream = TRUE) {
+
+  jsonlite::toJSON(list(inputs = list(list(id = "lat",
+                                           type = "text/plain",
+                                           value = p[2]),
+                                      list(id = "lon",
+                                           type = "text/plain",
+                                           value = p[1]),
+                                      list(id = "upstream",
+                                           type = "text/plain",
+                                           value = ifelse(upstream,
+                                                          "true", "false")))),
+                   pretty = TRUE, auto_unbox = TRUE)
+
+}
+
+make_json_input_xspt <- function(p, w, n) {
+
+  jsonlite::toJSON(list(inputs = list(list(id = "lat",
```

Comment:
Could use some help understanding / documenting units on these.

Reply:
These are both helpful questions to me.  Width is in meters.  lat and lon are decimal degrees.  As in my comments below I'll work on documentation and also figure out how to better document the post process metadata.  The Return is in meters.  The getxsatpoint since it is using NHDPlus specifically returns a 10m resolution cross-section (should it be 30? awk!).  The native function has an optional res parameter in it so we could add that. 
 getxsatendpts allows the user to specify the resolution.

Reply:
I think it would make some sense to add the same resolution attribute on this service.